Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml

```diff
@@ -0,0 +1,198 @@
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])    
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresultdetails_in
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser=(
+    starttime:datetime=datetime(null), 
+    endtime:datetime=datetime(null),
+    srcipaddr_has_any_prefix:dynamic=dynamic([]),
+    dstipaddr_has_any_prefix:dynamic=dynamic([]),
+    ipaddr_has_any_prefix:dynamic=dynamic([]), 
+    url_has_any:dynamic=dynamic([]),
+    httpuseragent_has_any:dynamic=dynamic([]),
+    eventresultdetails_in:dynamic=dynamic([]),
+    eventresult:string='*',
+    disabled:bool=false
+  )
+  {
+      let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let ip_any = set_union(srcipaddr_has_any_prefix, dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let EventLookup=datatable(DeviceAction:string, DvcAction:string,EventResult:string,HttpStatusCode:string)
+      [
+          "alert", "Allow", "Success","200"
+          , "allow", "Allow", "Success", "200"
+          , "continue", "Allow", "Success", "200"
+          , "override", "Allow", "Success", "200"
+          , "block-continue", "Allow", "Partial", "200"
+          , "block-url", "Deny", "Failure", "503"
+          , "block-override", "Deny", "Failure", "302"
+          , "override-lockout", "Deny", "Failure","503"
+          , "reset client", "Reset Source", "Failure", "503"
+          , "reset server", "Reset Destination", "Failure", "503"
+          , "reset both", "Reset", "Failure", "503"
+          , "deny", "Deny", "Failure", "503"
+          , "drop", "Drop", "Failure", "503"
+          , "drop ICMP", "Drop ICMP", "Failure", "503"
+      ];
+      let SeverityLookup=datatable(LogSeverity:string,EventSeverity:string)
+      [   1, "Informational" 
+          , 2, "Low" 
+          , 3, "Medium"
+          , 4, "Medium" 
+          , 5, "High"
+      ];
+      CommonSecurityLog
+      | where not(disabled)
+      | where DeviceVendor == "Palo Alto Networks"
+          and DeviceProduct == "PAN-OS"
+          and Activity == "THREAT"
+          and DeviceEventClassID == "url"
+      | where (isnull(starttime) or TimeGenerated >= starttime)
+        and (isnull(endtime) or TimeGenerated <= endtime)
+      | where (array_length(url_has_any) == 0 or RequestURL has_any (url_has_any))
```

Comment:
There is a challenge with the parameters including a protocol (http/https) and the events not. This is a simpler version of the big URL matching issue but can be solved here more simply. See what I did [here](https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimWebSession/Parsers/vimWebSessionzScalerZIA.yaml)

Reply:
Done